### ISSUE TITLE
[expo-av][iOS] Remote audio initial playing takes a long time on slow internet connection on iOS

### ISSUE THREAD

**Issue by serhiipalash** [expo/expo#5947]
### Environment

| software         | version
| ---------------- | -------
| expo | 35.0.0
| react-native     | https://github.com/expo/react-native/archive/sdk-35.0.0.tar.gz
| device    | iPhone 8 (simulator), iPhone 6s (device)
| platform | iOS

`exp diagnostics` output

<img width="1007" alt="Screen Shot 2019-10-11 at 2 24 11 PM" src="https://user-images.githubusercontent.com/14208343/66647904-d524f600-ec32-11e9-8e53-90c835f2b1b0.png">

### Steps to Reproduce

When you start remote audio playing on iOS, it takes a long time (7-10 sec) because on iOS actual playing starts only after initial buffering is finished and it takes 6-7 sec usually to buffer audio on start. Check this video and my iOS Simulator screenshot to see that `Playing` and `Buffered` have the same timestamp

https://drive.google.com/file/d/1y-F9aIrSpozatbsdJ3BtICey0uzbjdcc/view

![66649918-0ce26c80-ec38-11e9-8678-ed1378e63741 copy](https://user-images.githubusercontent.com/14208343/66650431-4798d480-ec39-11e9-9c48-d8734732e99e.png)

In opposite, on Android remote audio playing starts right after initial loading is finished and usually takes 1-2 sec. Check this screenshot from my Android device to see that `Playing` and `Loaded` have the same timestamp.

![66648719-e7a02f00-ec34-11e9-800e-65c089c83306 copy](https://user-images.githubusercontent.com/14208343/66650448-5089a600-ec39-11e9-8ea6-c16ee3dcd533.jpg)

For our app this bug is critical, even that it is only noticeable on slow internet connection. We have iTunes podcasts as content in our app and users often listen them in the road or subway, where internet connection could be very slow. Check our production bug video

https://drive.google.com/file/d/1gQWk-yCUhoDnmPjKJw_Lu6gkaWc3B9Jf/view

To reproduce this bug open this snack on iOS device and press "Play"

https://snack.expo.io/@serhiipalash/test-expo-audio

### Expected Behavior

When you start remote audio on iOS it plays right after initial loading and not after initial buffering.

### Actual Behavior

When you start remote audio on iOS it plays only after initial buffering is finished and it takes a long time.

### Reproducible Demo

Snack: https://snack.expo.io/@serhiipalash/test-expo-audio
Repo: https://github.com/serhiipalash/test-expo-audio

### Related issues
https://github.com/expo/expo/issues/5492

P.S. I saw that @sjchmiela submitted an issue https://github.com/expo/expo/issues/4459 about rewriting `expo-av` and it is in Expo 36 todos. I am not sure what do you plan to do with `expo-av`, but it would be great to have this issue fixed as it makes UX on slow connection horrible, when you try to play remote audio.

Thanks in advance!

**Comment by serhiipalash** [comment]
@cruzach I saw that you started to collect stories for [SDK 37 ](https://github.com/orgs/expo/projects/5), can you please add this bug to plan? :)

**Comment by mczernek** [comment]
I am afraid we cannot do much about that. I've spent some time debugging your Snack and, while I might be wrong, it seems to be just how iOS AV framework works.

Status callbacks are invoked, in most cases, whenever we got notified by platform on the native side (plus some cases which we add on our own). Values we pass are just values given to us by platform. There might (and are) some discrepancies between platforms we cannot really help. And it just seems to be how iOS handles such cases. Perhaps it waits for some larger chunk to be buffered than Android, before starting playback?
